fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/TransactionImpl.java

```diff
@@ -238,6 +255,12 @@ private void failIfRolledback() {
         }
     }
 
+    private void releaseLocks() {
+        execQuietly("Failed to release resource locks cleanly. You may need to restart Fedora.", () -> {
```

Comment:
Is this just going to be a log message? Seems like it could be a bigger problem or am I over estimating its severity?

Reply:
It is highly unlikely to happen. I'm not even sure how to engineer it to happen. But, yes, if locks were not released, then the resources that were not unlocked would remain locked until Fedora was restarted.